web/modules/weather_blocks/src/Plugin/Block/Test/EndToEnd/Observations/InvalidObservations.php.test

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace Drupal\weather_blocks\Plugin\Block\Test\EndToEnd;
+
+use Drupal\weather_blocks\Plugin\Block\CurrentConditionsBlock;
+
+/**
+ * Tests for current observations, invalid results.
+ */
+final class InvalidObservationsTest extends EndToEndBase
+{
+    /**
+     * Create the block under test.
+     */
+    protected function setUp($type = null): void
+    {
+        parent::setUp(CurrentConditionsBlock::class);
+    }
+
+    /**
+     * Tests that if the first observation is valid, we use it.
+     * @group e2e
+     */
+    public function testFirstObservationIsValid(): void
+    {
+        $this->onLocationRoute(33.211, -87.566);
+
+        $data = $this->block->build();
+
+        $this->assertSame(70, $data["temperature"]);
+    }
+
+    /**
+     * Tests that if the first observation is invalid, we use the second one.
+     * @group e2e
+     */
+    public function testFirstObservationIsInvalid(): void
+    {
+        $this->onLocationRoute(35.198, -111.651);
+
+        $data = $this->block->build();
+
+        $this->assertSame(64, $data["temperature"]);
+    }
+
```

Comment:
I thought about it, but I wasn't sure it gained us much confidence in the code. My gut feeling is that the only knew attestation would be *how* it iterates through observation stations, not that it actually does so since that's already attested?